### PR TITLE
Improved ListPage style

### PR DESCRIPTION
**Pull request by Harineko0**
## Summary

- Added SkeletonDocumentItem
- Added button.tsx and change List.tsx style
- Added menubar.tsx and change DocumentItem style
- Fix ViewHistoryServiceImpl.setViewHistory() to return updated viewHistory correctly
- Modify heading of List.tsx

## Summary by CodeRabbit

- **New Features**
  - Introduced a comprehensive `Menubar` component suite for building interactive menubars.
  - Added a `Skeleton` component for displaying loading placeholders.

- **Enhancements**
  - Simplified query keys in the view history API for better performance.
  - Conditional rendering in the `DocumentListSection` for improved document handling.

- **Bug Fixes**
  - Refactored `DocumentItem` to use new menubar components, enhancing UI consistency.

- **Dependencies**
  - Added `@radix-ui/react-menubar` package for advanced UI components.

- **Testing**
  - Removed unnecessary imports and declarations in the `ViewHistoryServiceImpl` test file.

### CODE REVIEW

File: one-pager-maker/src/api/viewHistoryApi.ts [hunk 1]
```diff
@@ -54,7 +54,15 @@ export const viewHistoryApi = {
             },
             onSuccess: async (history) => {
                 if (history === undefined) return;
-                await queryClient.refetchQueries({queryKey: [queryKeys.editHistoryId(history.id), queryKeys.editHistories]})
+                // queryClient.setQueryData([queryKeys.editHistoryId(history.id)], history);
```
Comment (binnmti) [inline one-pager-maker/src/api/viewHistoryApi.ts:57]: Did you forget to turn it off?

**Review by binnmti — APPROVED** [review]
LGTM

File: one-pager-maker/src/api/viewHistoryApi.ts (hunk 1 above)
Comment (Harineko0) [reply one-pager-maker/src/api/viewHistoryApi.ts]: I don't know what you mean, but I removed this comment `// queryClient.setQueryData([...`

Comment (binnmti) [reply one-pager-maker/src/api/viewHistoryApi.ts]: Did you forget to delete this comment? :)

**Pull request merged**